inference/server/oasst_inference_server/chat_repository.py

```diff
@@ -52,39 +53,60 @@ def create_chat(self) -> models.DbChatEntry:
     def add_prompter_message(self, chat_id: str, message_request: interface.MessageRequest) -> None:
         logger.info(f"Adding prompter message {message_request} to chat {chat_id}")
         chat = self.get_chat_by_id(chat_id, for_update=True)
-        if not chat.conversation.is_prompter_turn:
+        entry = chat.to_entry()
+        if not entry.conversation.is_prompter_turn:
             raise fastapi.HTTPException(status_code=400, detail="Not your turn")
         if chat.pending_message_request is not None:
             raise fastapi.HTTPException(status_code=400, detail="Already pending")
+        
+        entry.conversation.messages.append(protocol.ConversationMessage(
+            id=uuid.uuid4(),
+            text=message_request.message,
+            is_assistant=False,
+        ))
 
-        chat.conversation.messages.append(
-            protocol.ConversationMessage(
-                text=message_request.message,
-                is_assistant=False,
-            )
-        )
-
+        chat.conversation = protocol.Conversation(**entry.conversation.dict())
         chat.pending_message_request = message_request
         chat.message_request_state = interface.MessageRequestState.pending
         self.maybe_commit()
-        logger.debug(f"Added prompter message {message_request} to chat {chat_id}")
 
     def add_assistant_message(self, chat_id: str, text: str) -> None:
         logger.info(f"Adding assistant message {text} to chat {chat_id}")
         chat = self.get_chat_by_id(chat_id, for_update=True)
-        chat.conversation.messages.append(
-            protocol.ConversationMessage(
-                text=text,
-                is_assistant=True,
-            )
-        )
+        entry = chat.to_entry()
```

Comment:
Changed to update logic reference in the summary. The existing code doesn't actually persist the messages to the JSON field in the db. This fixes that by updating the root field with a new `protocol.Conversation` instance with the new message